master/buildbot/status/build.py

```diff
@@ -386,6 +374,27 @@ def updateBuildRequestIDs(self, brids):
     def setOwners(self, owners):
         self.owners = owners
 
+    @defer.inlineCallbacks
+    def setUserID(self, user_id):
+        self.user_id = user_id
+        if self.user_id:
+            return
+
+        # TODO: check in logz if "Found user_id in old way" is logged
+        # If yes, add generating user_id in that step
+        # Else remove below for-loop and tests for it.
+        for owner in self.owners:
```

Comment:
Right now is OK, but we must remember about deleting this code in next release because we can not unset user with this method. :) Please create a new task for it.

Reply:
https://jira.hq.unity3d.com/browse/CDSK-983